src/gui/pages_game/KM_GUIGameHouse.pas

```diff
@@ -891,6 +992,27 @@ procedure TKMGUIGameHouse.House_SchoolUnitChange(Sender: TObject; Shift: TShiftS
 end;
 
 
+{Toggle ware delivery for separate resources (wood, leather) in Armor workshop}
+procedure TKMGUIGameHouse.House_ArmorWSDeliveryToggle(Sender: TObject);
+var I: Byte;
+    WareType: TWareType;
+    ArmorWS: TKMHouseArmorWorkshop;
+begin
+  WareType := wt_None;

```

Comment:
It would be much simpler to assign buttons `Tag := I;` in creation and here just map from `const WORKSHOP_WARE: array [1..2] of TWareType = (wtWood, wtLeather); ..... WareType := WORKSHOP_WARE[Sender.Tag]`